pkg/spxrpc/place_stream_branding.go

```diff
@@ -0,0 +1,279 @@
+			Key:      key,
+			MimeType: mimeType,
+		}
+
+		// add dimensions if available
+		if width != nil {
+			w := int64(*width)
+			asset.Width = &w
+		}
+		if height != nil {
+			h := int64(*height)
+			asset.Height = &h
+		}
+
+		// for text assets, include data inline; for images, provide URL
+		if mimeType == "text/plain" {
+			str := string(data)
+			asset.Data = &str
+		} else {
+			url := fmt.Sprintf("/xrpc/place.stream.branding.getBlob?key=%s&broadcaster=%s", key, broadcasterID)
+			asset.Url = &url
+		}
+
+		assets = append(assets, asset)
+	}
+
+	return &placestreamtypes.BrandingGetBranding_Output{
+		Assets: assets,
+	}, nil
+}
+
+func (s *Server) isAdminDID(did string) bool {
+	for _, adminDID := range s.cli.AdminDIDs {
+		if adminDID == did {
+			return true
+		}
+	}
+	return false
+}
+
+func (s *Server) handlePlaceStreamBrandingUpdateBlob(ctx context.Context, input *placestreamtypes.BrandingUpdateBlob_Input) (*placestreamtypes.BrandingUpdateBlob_Output, error) {
+	// check authentication
+	session, _ := oatproxy.GetOAuthSession(ctx)
+	if session == nil {
+		return nil, echo.NewHTTPError(http.StatusUnauthorized, "oauth session not found")
+	}
+
+	// check admin authorization
+	if !s.isAdminDID(session.DID) {
+		log.Warn(ctx, "unauthorized branding update attempt", "did", session.DID)
+		return nil, echo.NewHTTPError(http.StatusUnauthorized, "not authorized to modify branding")
+	}
+
+	var broadcasterDID string
+	if input.Broadcaster != nil {
+		broadcasterDID = *input.Broadcaster
+	}
+	broadcasterID := s.getBroadcasterID(ctx, broadcasterDID)
+
+	// decode base64 data
+	data, err := base64.StdEncoding.DecodeString(input.Data)
+	if err != nil {
+		return nil, echo.NewHTTPError(http.StatusBadRequest, "invalid base64 data")
+	}
+
+	// validate size based on key type
+	maxSize := 500 * 1024 // 500KB default for logos
+	if input.Key == "favicon" {
+		maxSize = 100 * 1024 // 100KB for favicons
+	} else if input.Key == "siteTitle" || input.Key == "siteDescription" || input.Key == "primaryColor" || input.Key == "accentColor" || input.Key == "defaultStreamKey" || input.Key == "defaultStreamer" {
+		maxSize = 1024 // 1KB for text values
+	}
+	// sidebarBackgroundImage uses default 500KB limit
+	if len(data) > maxSize {
+		return nil, echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("blob too large (max %d bytes)", maxSize))
+	}
+
+	// store in database
+	var width, height *int
+	if input.Width != nil {
+		w := int(*input.Width)
+		width = &w
+	}
+	if input.Height != nil {
+		h := int(*input.Height)
+		height = &h
+	}
+
+	err = s.statefulDB.PutBrandingBlob(broadcasterID, input.Key, input.MimeType, data, width, height)
+	if err != nil {
+		log.Error(ctx, "failed to store branding blob", "err", err)
+		return nil, echo.NewHTTPError(http.StatusInternalServerError, "unable to store branding blob")
+	}
+
+	// invalidate cache
+	cacheKey := fmt.Sprintf("%s:%s", broadcasterID, input.Key)
+	s.BrandingCache.Delete(cacheKey)
```

Comment:
This part won't work multi-node. I wouldn't sweat a server-side cache at all honestly; there's a client-side cache for end users and the database is fast